akka-persistence-query/src/main/scala/akka/persistence/query/journal/leveldb/EventsByPersistenceIdPublisher.scala

```diff
@@ -181,7 +181,7 @@ private[akka] class CurrentEventsByPersistenceIdPublisher(
     deliverBuf()
     if (highestSeqNr < toSequenceNr)
       toSeqNr = highestSeqNr
-    if (highestSeqNr == 0L || (buf.isEmpty && currSeqNo > toSequenceNr))
+    if (highestSeqNr == 0L || (buf.isEmpty && currSeqNo > toSequenceNr) || currSeqNo == fromSequenceNr)
```

Comment:
Note that there is a `maxBufSize` which means that the buf might be half filled from a first replay request and then when this is called for the second replay request we must still drain the buffer before completing the stream. Shouldn't the condition be `buf.isEmpty && (currSeqNo > toSequenceNr || currSeqNo == fromSequenceNr)`?